src/structure_managers/property_block_sparse.hh

```diff
@@ -0,0 +1,600 @@
+    BlockSparseProperty(StructureManagerBase & manager,
+                        std::string metadata = "no metadata")
+        : Parent{manager, 0, 0, Order, PropertyLayer, metadata} {}
+
+    //! Default constructor
+    BlockSparseProperty() = delete;
+
+    //! Copy constructor
+    BlockSparseProperty(const BlockSparseProperty & other) = delete;
+
+    //! Move constructor
+    BlockSparseProperty(BlockSparseProperty && other) = default;
+
+    //! Destructor
+    virtual ~BlockSparseProperty() = default;
+
+    //! Copy assignment operator
+    BlockSparseProperty & operator=(const BlockSparseProperty & other) = delete;
+
+    //! Move assignment operator
+    BlockSparseProperty & operator=(BlockSparseProperty && other) = default;
+
+    /* ---------------------------------------------------------------------- */
+    //! return runtime info about the stored (e.g., numerical) type
+    const std::type_info & get_type_info() const final {
+      return typeid(Precision_t);
+    };
+
+    //! Adjust size so that each center are accessible
+    void resize() {
+      auto order = this->get_order();
+      auto new_size = this->base_manager.nb_clusters(order);
+      this->values.resize(new_size);
+      this->center_sizes.resize(new_size);
+      this->keys_list.resize(new_size);
+    }
+
+    template <size_t CallerLayer>
+    void initialize_to_zeros(const ClusterRefKey<Order, CallerLayer> & id,
+                             key_t & key) {
+      auto && index{id.get_cluster_index(CallerLayer)};
+      dense_t mat = dense_t::Zero(this->get_nb_row(), this->get_nb_col());
+      this->values[index][key] = mat;
+    }
+
+    size_t size() const { return this->values.size(); }
+
+    /**
+     * clear all the content of the property
+     */
+    void clear() {
+      this->values.clear();
+      this->all_keys.clear();
+      this->keys_list.clear();
+      this->center_sizes.clear();
+    }
+
+    /* ---------------------------------------------------------------------- */
+    //! Property accessor by cluster ref
+    template <size_t CallerLayer>
+    inline decltype(auto)
+    operator[](const ClusterRefKey<Order, CallerLayer> & id) {
+      static_assert(CallerLayer >= PropertyLayer,
+                    "You are trying to access a property that does not exist at"
+                    "this depth in the adaptor stack.");
+
+      return this->operator[](id.get_cluster_index(CallerLayer));
+    }
+
+    //! Accessor for property by cluster index and return a sparse
+    //! representation of the property associated to this cluster
+    inline input_data_t & operator[](const size_t & index) {
+      return this->values[index];
+    }
+
+    inline size_t get_dense_feature_size(const size_t & index) {
+      auto keys = this->values[index].get_keys();
+      return this->get_nb_comp() * keys.size();
+    }
+
+    //! Accessor for property by cluster index and return a dense
+    //! representation of the property associated to this cluster
```

Comment:
Representation is the right word in this context, but given our use or representation, we should probably use something else? Or do you think this is unambiguous?